### PR TITLE
[14.0] [FIX] web_advanced_search: Many2one selection on Filters missing

### LINKED ISSUE #1957: [14.0] web_advanced_search Many2one selection on Filters missing
Looks like they didn't migrate this part to Odoo14 because it wasn't super clear from the comment where it was used. https://github.com/OCA/web/pull/1909#issuecomment-828997442 

The functionality this enabled in v13 is to have a many2one dropdown when adding a filter for a many2one field.

### PR DESCRIPTION
**Pull request by jeroenev**
Includes code for searching functionality on relational fields when adding filters.

Fixes: #1957

### CODE REVIEW

File: web_advanced_search/static/src/xml/web_advanced_search.xml [hunk 1]
```diff
@@ -1,13 +1,29 @@
 <?xml version="1.0" encoding="UTF-8" ?>
-<!-- Copyright 2017-2018 Jairo Llopis <jairo.llopis@tecnativa.com>
```
Comment (pedrobaeza) [inline web_advanced_search/static/src/xml/web_advanced_search.xml]: Please don't remove copyright, but add yours if needed

**Review by pedrobaeza — COMMENTED** [review]
Thanks for restoring back this!

Several comments:

- On relational fields, it would be good to autoselect "Is equal to" operator by default.
- Using up and down arrows for moving between results on what you have typed makes the dropdown to be closed. I suppose any prevent propagation is missing.
- What about adding the feature on the popup (domain widget)?

**Comment by jeroenev** [comment]
> On relational fields, it would be good to autoselect "Is equal to" operator by default.

I had this at first, but It would keep the data from the previous field if I switched to a different model
So if I had Contact selected, and then switch to a Company field, it would continue showing contacts in the widget

I tried using a custom `ComponentAdapter` to replace or update the `FieldWidget`, and tried adding the `FieldWidget` as a prop, but I couldn't get it working properly without errors using `ComponentAdapter` / `renderWidget` / `updateWidget`.
So I removed the code for auto-selecting `==`, so I could make it work reliably.

Probably overlooked something, but couldn't find what exactly. 

The other 2 things I can check later, at least the 'basic' functionality works now

**Comment by pedrobaeza** [comment]
OK, thanks for the effort. When my colleague comes from holidays, he will check the technical part.

**Review by NL66278 — APPROVED** [review]
:+1: This is definitely a nice improvement and when tested worked well.

I am currently also trying to get some enhancements to work. Especially restoring the old functionality to have an "in selection" operator, where for x2x fields you can open a tree of a related model, enter search criteria, or select records, and then apply either the search criteria on the related model, or use the records selected... Have to learn a lot about how to interact with Owl. This PR provides some welcome pointers.

**Comment by pedrobaeza** [comment]
@NL66278 I don't see such feature as needed if we improve the domain widget for accepting is equal + many2one input. Is there anything missing from this widget?

**Comment by NL66278** [comment]
@pedrobaeza The original web_advanced_search_x2x made it very easy to enter advanced search criteria from the actual tree view + search of the related model. And to just select a few (or a lot) of records in the related model and say the related model should be in this list of records. I think with some work, you can get a part of this functionality from the domain widget -and it is great to have this widget at least, but I think not for all users this will be easy to do.

**Comment by pedrobaeza** [comment]
True, selecting specific records from previous filtering is not available here, but we have it through https://github.com/OCA/web/tree/13.0/web_widget_domain_editor_dialog, so we can make a mention in the README and to not develop anything more. Check it and tell me what you think about it.

**Comment by NL66278** [comment]
@pedrobaeza I will check it out. Thanks for the pointer!

**Review by rousseldenis — APPROVED** [review]

**Comment by NL66278** [comment]
@pedrobaeza I checked out the web_widget_domain_editor_dialog. Definitely a nice module, with great functionality. But I think it is a little bit more geared to administrators / designers then to the educated end-users. You have to enter debug mode, manually select the appropriate model etc. Not a problem for you or me... but might be a hurdle for end-users / application managers. And I did not immediately see an equivalent for picking a number of records in a related model, but maybe I did not venture deep enough. Anyhow, this is my proposal to restore the old 'is in selection' logic: https://github.com/OCA/web/pull/2033

**Comment by spadae22** [comment]
@pedrobaeza HI, I am having this module updated to match and improve searches for administrators / designers like myself as I used this module in v8 forever.  Thanks to all for comments and contributions. I am just writing the check to contribute back to the OCA.

**Pull request merged**